File: InterestingParty.php
<?php
class InterestingParty
{
    public function __construct()
    {
    }

    public function bestInvitation(Array $first, Array $second)
    {
        $interest_count = array();
        for ($i = 0; $i < count($first); $i++) {
            $this->countup($interest_count, $first[$i]);
            $this->countup($interest_count, $second[$i]);
        }
        return max($interest_count);
    }

    private function countup(&$interest_count, $interest)
    {
        if (@$interest_count[$interest] === null) $interest_count[$interest] = 0;
        $interest_count[$interest]++;
    }
}

File: InterestingParty.rb
class InterestingParty

  def initialize
    @invitation_count = {}
  end

  def bestInvitation(first, second)
    countup(first + second)
    k, v = @invitation_count.max { |a, b| a[1] <=> b[1] }
    v
  end

  def countup(invitations)
    invitations.each do |i|
      @invitation_count[i] = 0 if @invitation_count[i] == nil
      @invitation_count[i] += 1
    end
  end
end
